Tribler/community/multichain/community.py

```diff
@@ -282,6 +289,93 @@ def get_statistics(self, public_key=None):
             statistics["total_down"] = 0
         return statistics
 
+    @blocking_call_on_reactor_thread
+    def get_graph(self, public_key=None, neighbor_level=1):
+        """
+        Return a dictionary with the neighboring nodes and edges of a certain focus node within a certain radius,
+            regarding the local multichain database.
+
+        :param public_key: the public key of the focus node
+        :param neighbor_level: the radius within which the neighbors have to be returned
+        :return: a tuple of a dictionary with nodes and a dictionary with edges
+        """
+        if public_key is None:
+            public_key = self.my_member.public_key
+        list_of_nodes = self.get_list_of_nodes(public_key, neighbor_level)
+        nodes = []
+        for current_key in list_of_nodes:
+            nodes.append({"public_key": current_key, "total_up": self.database.total_up(current_key),
```

Comment:
For now this is probably fine, but please add a `TODO` for retrieving this data all at once with a query instead of doing all these small queries.